### PR TITLE
Fix parsing of `PARTITION BY KEY`

### PR DESCRIPTION
**Pull request by dbussink**
The `PARTITION BY KEY` syntax allows for passing in an empty column list which signifies that the primary key is used for partitioning.

This needs to be added to the parser to support this case.

See also https://dev.mysql.com/doc/refman/8.0/en/partitioning-key.html, specifically:

```
KEY takes only a list of zero or more column names. Any columns used as the partitioning key must
comprise part or all of the table's primary key, if the table has one. Where no column name is
specified as the partitioning key, the table's primary key is used, if there is one.
```

## Related Issue(s)

Found as part of #10203

Fixes https://github.com/vitessio/vitess/issues/10959

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -4778,6 +4778,19 @@ column_list_opt:
...
+    $$ = nil
+  }
+| sql_id
+  {
+    $$ = Columns{$1}
+  }
+| column_list_empty ',' sql_id
+  {
+    $$ = append($$, $3)
+  }
```
Comment (GuptaManan100) [inline go/vt/sqlparser/sql.y:4792]: Why not just use column_list here?

Comment (dbussink) [reply go/vt/sqlparser/sql.y:4792]: Good point, that's nicer indeed. Let me fix that up.

Comment (dbussink) [reply go/vt/sqlparser/sql.y]: @GuptaManan100 Cleaned this up!

**Comment by GuptaManan100** [comment]
@dbussink Do you mind creating an issue for the bug as well?

**Review by GuptaManan100 — APPROVED** [review]

**Comment by shlomi-noach** [comment]
A new day, a new MySQL parser syntax surprise.

**Pull request merged**